forc-pkg/src/pkg.rs

```diff
@@ -2774,9 +2814,54 @@ mod test {
         .unwrap()
     }
 
+    #[test]
+    fn test_collect_test_entry_logs() {
+        let current_dir = env!("CARGO_MANIFEST_DIR");
+        let manifest_dir = format!("{current_dir}/tests/test_pkg_entry");
+        let build_opts = BuildOpts {
+            pkg: PkgOpts {
+                path: Some(manifest_dir),
+                ..Default::default()
+            },
+            build_target: BuildTarget::Fuel,
+            tests: true,
+            experimental: ExperimentalFlags { new_encoding: true },
+            ..Default::default()
+        };
+        let built_pkgs = build_with_options(build_opts).unwrap();
+        let built_pkg = built_pkgs.expect_pkg().unwrap();
+        let test_entry = built_pkg
+            .bytecode
+            .entries
+            .iter()
+            .find_map(|entry| {
+                if let PkgEntryKind::Test(test_entry) = &entry.kind {
+                    Some(test_entry)
+                } else {
+                    None
+                }
+            })
+            .unwrap();
+
+        let log_params: Vec<_> = test_entry
+            .log_params
+            .iter()
+            .map(|(_, type_info)| type_info)
+            .collect();
+        assert_eq!(log_params.len(), 1);
+        let type_info = log_params[0].clone().unwrap();
+
+        assert!(matches!(
+            type_info.as_ref(),
+            TypeInfo::UnsignedInteger(IntegerBits::SixtyFour)
```

Comment:
Is it working for other types? If so we should test them (other numeric types, bool, string, array, tuple, struct, enum).

Reply:
Yes we should add more test but i think exhaustively trying all the types would go out of scope because:

1. imo at that point we are testing type engine (which should be tested on its own) not this function, as types itself are not coming from this new function but from type engine itself
2. There are a lot of types possible :)

I agree that it makes sense to add more tests that, maybe we can start with one that collects multiple logs that are in different types so that we can test if we can collect them in order?

Reply:
On a second thought I am thinking about adding similar tests for:

1. other numeric types, 
2. bool,
3. string,
4. array,
5. tuple, 
6. struct,
7.  enum,

and a function based one that returns one of these types
and one that returns multiple log types

I think this is not exhaustive but common types that will be encountered with logs, also it will make sure that we don't rely on type engine tests as that might not be exhaustive either.

Reply:
Having a test for multiple log types is a good idea! I agree it doesn't need to be exhaustive. It's good to test as much as we can within reason to find bugs that would otherwise surface to users of forc-test.